pkg/gui/mergeconflicts/state.go

```diff
@@ -11,42 +11,62 @@ type Selection int
 
 const (
 	TOP Selection = iota
+	MIDDLE
 	BOTTOM
 	BOTH
 )
 
-// mergeConflict : A git conflict with a start middle and end corresponding to line
+// mergeConflict : A git conflict with a start, ancestor (if exists), target, and end corresponding to line
 // numbers in the file where the conflict markers appear
 type mergeConflict struct {
-	start  int
-	middle int
-	end    int
+	start    int
+	ancestor int
+	target   int
+	end      int
 }
 
 type State struct {
 	sync.Mutex
-	conflictIndex int
-	conflictTop   bool
-	conflicts     []*mergeConflict
-	EditHistory   *stack.Stack
+	conflictIndex     int
+	conflictSelection Selection
+	conflicts         []*mergeConflict
+	EditHistory       *stack.Stack
 }
 
 func NewState() *State {
 	return &State{
-		Mutex:         sync.Mutex{},
-		conflictIndex: 0,
-		conflictTop:   true,
-		conflicts:     []*mergeConflict{},
-		EditHistory:   stack.New(),
+		Mutex:             sync.Mutex{},
+		conflictIndex:     0,
+		conflictSelection: TOP,
+		conflicts:         []*mergeConflict{},
+		EditHistory:       stack.New(),
 	}
 }
 
-func (s *State) SelectTopOption() {
-	s.conflictTop = true
+func (s *State) SelectPrevConflictHunk() {
+	switch s.conflictSelection {
```

Comment:
I think this might be easier to read if we had a method returning a slice of the available states e.g. [TOP, MIDDLE, BOTTOM]' and then based the logic off of that slice.

Reply:
@jesseduffield Is it correct that what you mean is like the following?:

```go
func (c *mergeConflict) availableSelections() []Selection {
	if c.hasAncestor() {
		return []Selection{TOP, MIDDLE, BOTTOM}
	} else {
		return []Selection{TOP, BOTTOM}
	}
}
```